src/hooks/index.ts

```diff
@@ -1,110 +1,69 @@
  */
 export function useInactiveListener(suppress = false) {
-  const { active, error, activate, deactivate } = useWeb3ReactCore() // specifically using useWeb3React because of what this hook does
+  // const { activate, deactivate } = useWeb3ReactCore() // specifically using useWeb3React because of what this hook does
+  const { active, eipProvider, library } = useActiveWeb3React()
+  // const [ {chains, connectedChain, settingChain }, setChain] = useSetChain()
+  const [ activated, setActivated] = useState(false)
   const metaMaskInfo = useMetaMask()
-  const { ethereum } = window
+  // const {ethereum} = window // eipProvider used instead of ethereum as not all supported wallets use the ethereum namespace
+
+  const [primaryWallet] = onboard.state.get().wallets
     useEffect(() => {
-        if (ethereum && !active && !error && !suppress) {
-          // todo: add activators for coinbase
-            const handleChainChanged = () => {
-                // eat errors
-                activate(injected, undefined, true).catch(error => {
-                    console.error('Failed to activate after chain changed', error)
-                })
-                // .then(() => window.location.reload()) // suggested by MetaMask Docs
-            } 
+      if (active && eipProvider && !activated){
 
-            const handleAccountsChanged = (accounts: string[]) => {
-                if (accounts.length > 0) {
-                    // eat errors
-                    activate(injected, undefined, true).catch(error => {
-                        console.error('Failed to activate after accounts changed', error)
-                    })
-                } else {
-                  deactivate()
-                  if (metaMaskInfo.isMultiple) {
-                    window.location.reload()
-                  }
-                }
-            }
+        const handleChainChanged = () => {
+          // console.log("chainchanged")
+          OnboardSubscribe()
+        }
 
-            if (metaMaskInfo.isMultiple && ethereum.selectedProvider?.on) {
-              ethereum.selectedProvider.on('chainChanged', handleChainChanged) 
-              ethereum.selectedProvider.on('accountsChanged', handleAccountsChanged)  
-            } else if (ethereum.on){
-              ethereum.on('chainChanged', handleChainChanged)
-              ethereum.on('accountsChanged', handleAccountsChanged)
+        const handleAccountsChanged = async(accounts: string[]) => {
+          // console.log('accounts changed')
+          // console.log('accounts changed accounts -->', {accounts})
+            if (accounts.length > 0) {
+              OnboardSubscribe()
+            } else {
+              await onboard.disconnectWallet({label: primaryWallet.label})
+              if (metaMaskInfo.isMultiple || primaryWallet.label === 'WalletConnect') {
+                window.location.reload()
+              }
             }
+        }
 
+        // console.log({metaMaskInfo})
```

Comment:
you can use onboard hooks to handle accountsChanged and chainChanged all in one single file together where you gonna save the wallet state into localstorage